### PR TITLE
fedekunze/1585-added LiCoin for PageWallet

### LINKED ISSUE #1585: Create LiCoin for wallet page
Description:

We could use a slight modification of `TmLiTransaction` to create a better looking coin for `PageWallet` based on #1424

### LINKED ISSUE #1612: Define props types
Description:

Required by linter otherwise returns a warning

### PR DESCRIPTION
**Pull request by fedekunze**
Closes #1585
Closes #1612

Screenshot:

screen shot 2018-11-22 at 1 24 30 pm

❤️ Thank you!

---

* [x] Added entries in `CHANGELOG.md` with issue # and GitHub username
* [x] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

File: app/src/renderer/scripts/axiosProxy.js [hunk 1]
```diff
@@ -13,21 +13,15 @@ module.exports = (requestCounter = 0) => {
 
       const channel = `Axios/${requestCounter}`
 
-      console.log(`Request ${options.method.toUpperCase()} ${options.url}`)
```
Comment (faboweb) [inline app/src/renderer/scripts/axiosProxy.js]: Duuuuude, I commented now 3 times on this line. ^^ Why do you want to remove it so bad?

Comment (fedekunze) [reply app/src/renderer/scripts/axiosProxy.js]: hahaha bc it's annoying since it spams the console and also bc I can't get the actual requests on the Network tab of the Inspector

File: app/src/renderer/components/wallet/LiCoin.vue [hunk 2]
```diff
@@ -0,0 +1,88 @@
+<template lang="pug">
+.li-coin
```
Comment (jbibla) [inline app/src/renderer/components/wallet/LiCoin.vue]: make it an `<li>`

File: app/src/renderer/components/wallet/PageWallet.vue [hunk 3]
```diff
@@ -27,14 +27,12 @@ tm-page(data-title="Wallet")
         a(href="https://gaia.faucetcosmos.network/") token faucet
         | &nbsp;to aquire tokens to play with.
     data-empty-search(v-else-if="filteredBalances.length === 0")
-    tm-list-item.tm-li-balance(
+    li-coin.tm-li-balance(
```
Comment (jbibla) [inline app/src/renderer/components/wallet/PageWallet.vue]: put inside `<ul>` or `<ol>`

File: app/src/renderer/components/wallet/PageWallet.vue [hunk 4]
```diff
@@ -27,14 +27,12 @@ tm-page(data-title="Wallet")
...
       v-for="i in filteredBalances"
```
Comment (jbibla) [inline app/src/renderer/components/wallet/PageWallet.vue]: can we use `denom` instead of `i`?

File: app/src/renderer/components/wallet/LiCoin.vue (hunk 2 above)
Comment (jbibla) [inline app/src/renderer/components/wallet/LiCoin.vue]: @fedekunze what is your thinking re: `coin` vs. `token` or `denom`?

File: app/src/renderer/components/wallet/PageWallet.vue [hunk 5]
```diff
@@ -129,13 +129,13 @@ export default {
 </script>
 
 <style lang="stylus">
-main
-  .tm-li-label
-    max-width calc(100% - 110px)
+@require '~variables'
+
+@media screen and (min-width: 640px)
```
Comment (jbibla) [inline app/src/renderer/components/wallet/PageWallet.vue]: remove media queries

File: app/src/renderer/scripts/axiosProxy.js (hunk 1 above)
Comment (faboweb) [reply app/src/renderer/scripts/axiosProxy.js]: > I can't get the actual requests on the Network tab of the Inspector

yeah that is over for now since we need to proxy the request to the main thread :( will be back when we are on the web

> it's annoying since it spams the console

do you have an alternative? I would like to:
- see when a request is dispatched
- see when a request fails
- see when a request is successful

do we not need any of those?

File: app/src/renderer/components/wallet/LiCoin.vue (hunk 2 above)
Comment (fedekunze) [reply app/src/renderer/components/wallet/LiCoin.vue]: My order of preferences:
`LiCoin` > `LiCoinBalance` > `LiToken`

File: app/src/renderer/scripts/axiosProxy.js (hunk 1 above)
Comment (fedekunze) [reply app/src/renderer/scripts/axiosProxy.js]: > yeah that is over for now since we need to proxy the request to the main thread :( will be back when we are on the web

Oh ok then I guess we'll have to leave them like that

File: app/src/renderer/components/wallet/LiCoin.vue (hunk 2 above)
Comment (jbibla) [reply app/src/renderer/components/wallet/LiCoin.vue]: can you explain your thinking a little bit? to my knowledge, this is the first time we use `coin` anywhere - usually we use `denom` - `token` would also be suitable in my mind.

File: app/src/renderer/components/wallet/PageWallet.vue (hunk 5 above)
Comment (jbibla) [reply app/src/renderer/components/wallet/PageWallet.vue]: i know it's not the scope of this ticket but since we already know this is the plan - we might as well remove them as opposed to editing them IMO

File: app/src/renderer/components/wallet/LiCoin.vue (hunk 2 above)
Comment (fedekunze) [reply app/src/renderer/components/wallet/LiCoin.vue]: you mean `li.li-coin` ?

Comment (fedekunze) [reply app/src/renderer/components/wallet/LiCoin.vue]: SDK uses the struct `Coin`, which has a `denom` (denomination) and `amount`. I prefer to stick with Coin for consistency

File: app/src/renderer/components/wallet/PageWallet.vue (hunk 3 above)
Comment (fedekunze) [reply app/src/renderer/components/wallet/PageWallet.vue]: does it make any difference here ?

File: app/src/renderer/components/wallet/PageWallet.vue (hunk 4 above)
Comment (fedekunze) [reply app/src/renderer/components/wallet/PageWallet.vue]: switching it to `coin`

File: app/src/renderer/components/wallet/LiCoin.vue (hunk 2 above)
Comment (jbibla) [reply app/src/renderer/components/wallet/LiCoin.vue]: yes

File: app/src/renderer/components/staking/LiValidator.vue [hunk 6]
```diff
@@ -60,8 +60,8 @@ export default {
       return this.validator.revoked
         ? `Revoked`
         : this.validator.isValidator
-        ? `Validator`
-        : `Candidate`
+          ? `Validator`
```
Comment (jbibla) [inline app/src/renderer/components/staking/LiValidator.vue]: this is odd - what's going on here?

File: app/src/renderer/components/wallet/PageSend.vue [hunk 7]
```diff
@@ -2,59 +2,59 @@
...
+        tm-form-msg(name="Amount" type="between" :min="max ? 1 : 0" :max="max"
+          v-if="!$v.fields.amount.between")
 
-      p(v-if='mockedConnector')
+      p(v-if="mockedConnector")
         span Try sending to the address "
         strong(style="font-weight: bold") cosmos1p6zajjw6xged056andyhn62lm7axwzyspkzjq0
-        span ", it's a friendly bot which will send the money back to you!
-      br(v-if='mockedConnector')
+        span ", it"s a friendly bot which will send the money back to you!
```
Comment (jbibla) [inline app/src/renderer/components/wallet/PageSend.vue]: what's up with the quotes in this file? this one is a double quote?

File: app/src/renderer/components/staking/LiValidator.vue (hunk 6 above)
Comment (fedekunze) [reply app/src/renderer/components/staking/LiValidator.vue]: Idk those are changes from the linter

File: app/src/renderer/components/wallet/PageSend.vue (hunk 7 above)
Comment (fedekunze) [reply app/src/renderer/components/wallet/PageSend.vue]: Fixed. It was bc my linter suggested to change them to double quotes to see the colors from the pug file lol

**Comment by fedekunze** [comment]
I updated the test according to the warnings, should I split this PR ?

File: test/unit/specs/components/common/TmBalance.spec.js [hunk 8]
```diff
@@ -31,12 +36,10 @@ describe(`TmBalance`, () => {
   it(`shows correct stats depending on props`, () => {
     expect(wrapper.contains(`.unbonded-atoms`)).toBe(true)
 
-    expect(wrapper.contains(`.total-earnings`)).toBe(false)
```
Comment (faboweb) [inline test/unit/specs/components/common/TmBalance.spec.js]: why did you remove those checks?

Comment (fedekunze) [reply test/unit/specs/components/common/TmBalance.spec.js]: they required the props to be set

Comment (fedekunze) [reply test/unit/specs/components/common/TmBalance.spec.js]: and it was already tested on line 41

Comment (faboweb) [reply test/unit/specs/components/common/TmBalance.spec.js]: hmm, I think not.
```
expect(wrapper.contains(`.total-earnings`)).toBe(false)
```
is the negative test (it not showing yet)

```
expect(wrapper.contains(`.total-earnings`)).toBe(true)
```
is the positive test (now it is showing)

Maybe set the `totalEarnings` to 0 before?

Comment (faboweb) [reply test/unit/specs/components/common/TmBalance.spec.js]: 👍

File: app/src/renderer/components/governance/ModalDeposit.vue [hunk 9]
```diff
@@ -54,7 +54,20 @@ export default {
   directives: {
     ClickOutside
   },
-  props: [`proposalId`, `proposalTitle`, `denom`],
+  props: {
+    proposalId: {
+      type: [Number, String],
```
Comment (jbibla) [inline app/src/renderer/components/governance/ModalDeposit.vue]: why both? let's pick one?

File: app/src/renderer/scripts/axiosProxy.js [hunk 10]
```diff
@@ -12,18 +12,17 @@ module.exports = (requestCounter = 0) => {
       }
 
       const channel = `Axios/${requestCounter}`
-
-      console.log(`Request ${options.method.toUpperCase()} ${options.url}`)
       ipcRenderer.once(channel, (event, { exception, value }) => {
         ipcRenderer.removeAllListeners(channel)
-
         if (exception) {
+          // TODO: delete console logs on web
```
[the remark this answers is not in the document]
Comment (jbibla) [reply app/src/renderer/scripts/axiosProxy.js]: gettin' ready for the web 🎉

**Comment by jbibla** [comment]
this is great @fedekunze ! please remove the "warn" from package.json!
